### PR TITLE
HPCC-15124 Workunit scope checking issues with Cassandra WU store

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@cloLN Can you test please

**Comment by cloLN** [comment]
looks to be working.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -140,17 +140,16 @@ static bool checkWuSecAccess(IConstWorkUnit &cw, ISecManager *secmgr, ISecUser *
...
 static bool checkWuSecAccess(const char *wuid, ISecManager *secmgr, ISecUser *secuser, int required, const char *action, bool excpt, bool log)
 {
-    StringBuffer wuRoot;
-    Owned<IRemoteConnection> conn = querySDS().connect(getXPath(wuRoot, wuid).str(), myProcessSession(), 0, SDS_LOCK_TIMEOUT);
-    if (conn)
+    if (!secmgr || !secuser)
+        return true;
+    Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+    Owned<IConstWorkUnit> cw = factory->openWorkUnit(wuid);
+    bool ret=secmgr->authorizeEx(RT_WORKUNIT_SCOPE, *secuser, cw->queryWuScope())>=required;
```
Comment (ghalliday) [inline common/workunit/workunit.cpp]: could code as checkWuScopeSecAccess(cw->queryWuScope(), secmg, secuser...)

**Pull request merged**